packages/iTwinUI-react/src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -18,6 +18,10 @@ export type MiddleTextTruncationProps = {
    * @default 6
    */
   endCharsCount?: number;
+  /**
+   * Callback fired when text is truncated.
+   */
+  onTruncate?: (isTruncated: boolean) => void;
```

Comment:
Just to be sure that this api will be fine and extendable, could you update the existing middle text truncation story in `Select`. It is under our apps/Storybook.
I am also thinking passing children instead of listening to callback might be an option too.